lib/commonTypes.ts

```diff
@@ -33,3 +35,52 @@ export interface GetDaoProposalsConfig {
    */
   proposalId?: Hash;
 }
+
+export enum SchemePermissions {
+  None = 0,
+  /**
+   * A scheme always automatically gets this bit when registered to a DAO
+   */
+  IsRegistered = 1,
+  CanRegisterSchemes = 2,
+  CanAddRemoveGlobalConstraints = 4,
+  CanUpgradeController = 8,
+  CanCallDelegateCall = 0x10,
+  All = 0x1f,
+}
+/* tslint:disable:no-bitwise */
+/* tslint:disable:max-line-length */
+/**
+ * These are the permissions that are the minimum that each scheme must have to
+ * be able to perform its full range of functionality.
+ *
+ * Note that '1' is always assigned to a scheme by the Controller when the
+ * scheme is registered with the controller.
+ */
+export enum DefaultSchemePermissions {
+  NoPermissions = SchemePermissions.None,
+  MinimumPermissions = SchemePermissions.IsRegistered,
+  AllPermissions = SchemePermissions.All,
+  ContributionReward = SchemePermissions.IsRegistered,
```

Comment:
do we really need this enum? seems like we could define the defaults in each contractwrapper class